packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -416,6 +417,66 @@ const forwardPred = (ref: LocalReferencePosition): boolean =>
 const backwardPred = (ref: LocalReferencePosition): boolean =>
 	ref.slidingPreference === SlidingPreference.BACKWARD;
 
+class Obliterates {
+	/**
+	 * Array containing the all move operations within the
+	 * collab window
+	 *
+	 * When a segment is inserted, we must traverse to the left and right of it
+	 * to determine whether the segment was inserted into an obliterated range.
+	 * By keeping track of all move seqs, we can significantly reduce the search
+	 * space we must traverse.
+	 *
+	 * Sequence numbers in `moveSeqs` are sorted to accelerate bookkeeping.
+	 *
+	 * See https://github.com/microsoft/FluidFramework/blob/main/packages/dds/merge-tree/docs/Obliterate.md#remote-perspective
+	 * for additional context
+	 */
+	private readonly seqOrdered = new DoublyLinkedList<ObliterateInfo>();
+
+	private readonly startOrdered = new SortedSegmentSet<LocalReferencePosition>();
+
+	constructor(private readonly mergeTree: MergeTree) {}
+
+	public setMinSeq(minSeq: number): void {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+		while (!this.seqOrdered.empty && this.seqOrdered.first?.data.seq! <= minSeq) {
```

Comment:
this now does nothing if there are not obliterates, and it's at most O(n) for the number of obliterates

Reply:
If we're at the minSeq, does that mean that the obliterate should be considered out of the collabWindow? I had been assuming that it was the opposite (===minSeq means still inside collabWindow).